iFixit/checkout-ui-extensions
# Reorganize extension targets

This change reorganizes the entry points of the app to reflect the extension targets they use. This makes it easier to visually validate that the toml configuration is correct and that a given extension is targeting the right location in the UI.

This also upgrades the api version from 2023-07 to 2024-01. There are no breaking changes for [2023-10](https://shopify.dev/docs/api/release-notes/2023-10) or [2024-01](https://shopify.dev/docs/api/release-notes/2024-01).

The documentation [illustrates](https://shopify.dev/docs/apps/checkout/best-practices/multi-page#pre-and-post-purchase) that we can specify named exports for rendering multiple extension targets from the same module. We could consider using this strategy for the `order-status` and `thank-you` targets, since they are currently the same. I personally think we should implement more features first, and later if we identify that they have no differences we can merge them.

## QA

Make sure the app still builds with `pn dev`. As long as it builds, then all the extension points should be configured properly. An error would look like this:

![image](https://github.com/iFixit/checkout-ui-extensions/assets/52104630/821da106-7951-42a7-a0f1-eee29b00b1af)
